metadoc-js/src/main/scala/metadoc/ScalaReferenceProvider.scala

```diff
@@ -0,0 +1,47 @@
+package metadoc
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import scala.meta._
+import metadoc.schema.{Index, Symbol, Position}
+import monaco.{CancellationToken, Position => EditorPosition, Range, Uri}
+import monaco.editor.IReadOnlyModel
+import monaco.languages.{Location, ReferenceContext, ReferenceProvider}
+
+@ScalaJSDefined
+class ScalaReferenceProvider(attrs: Attributes, index: Index) extends ReferenceProvider {
+  override def provideReferences(
+    model: IReadOnlyModel,
+    position: EditorPosition,
+    context: ReferenceContext,
+    token: CancellationToken) = {
+
+    val offset = model.getOffsetAt(position)
+
+    def resolveLocation(pos: Position) = {
+      val startPos = model.getPositionAt(pos.start)
+      val endPos = model.getPositionAt(pos.end)
+      val range = new Range(
+        startPos.lineNumber, startPos.column,
+        endPos.lineNumber, endPos.column
+      )
+
+      new Location(model.uri, range)
+    }
+
+    val symbols: Seq[String] =
+      attrs.names.collectFirst {
+        case (pos, sym) if pos.start.offset <= offset && offset <= pos.end.offset =>
```

Comment:
Sounds good 👍 the tests can be written in `metadoc-tests`, where the fixture with `example` is setup.